_posts/2022-07-20-#25218.md

```diff
@@ -7,17 +7,52 @@ authors: [furszy]
 components: ["refactoring"]
 host: stickies-v
 status: upcoming
-commit:
+commit: 111ea3a
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- To simplify error handling in the [call site](https://en.wikipedia.org/wiki/Call_site), functions in Bitcoin Core quite often return a boolean to indicate if the operation was successful. With that pattern, one or multiple [out-parameters](http://www.cs.ecu.edu/karl/2530/spr18/Notes/lec21A.html#logical-calling-modes:~:text=an%20implicit%20one.-,Logical%20calling%20modes,-Call%20by%20value) can be used to access the data that was modified or created within the function, as well as to access information about how the operation happened (e.g. error messages).
```

Comment:
```suggestion
- To simplify error handling in the [call site](https://en.wikipedia.org/wiki/Call_site), functions in Bitcoin Core quite often return a boolean to indicate if the operation was successful, along with one or multiple [out-parameters](http://www.cs.ecu.edu/karl/2530/spr18/Notes/lec21A.html#logical-calling-modes:~:text=an%20implicit%20one.-,Logical%20calling%20modes,-Call%20by%20value) to access the function result and information about the operation, e.g. error messages.
```

Reply:
Simplification here makes sense - just modified your suggestion a bit to not have "along with" to highlight that out-parameters are not being "returned".

> To simplify error handling in the [call site](https://en.wikipedia.org/wiki/Call_site), functions in Bitcoin Core quite often return a boolean to indicate if the operation was successful. One or multiple [out-parameters](http://www.cs.ecu.edu/karl/2530/spr18/Notes/lec21A.html#logical-calling-modes:~:text=an%20implicit%20one.-,Logical%20calling%20modes,-Call%20by%20value) can then be used to access the function result and information about the operation (e.g. error messages).